tests/conftest.py

```diff
@@ -1771,12 +1780,25 @@ def core_dump_and_config_check(duthosts, request):
                 pre_existing_core_dumps = duthost.shell('ls /var/core/')['stdout'].split()
             duts_data[duthost.hostname]["pre_core_dumps"] = pre_existing_core_dumps
 
+            logger.info("Collecting running config before test on {}".format(duthost.hostname))
+            duts_data[duthost.hostname]["pre_running_config"] = {}
             if not duthost.stat(path="/etc/sonic/running_golden_config.json")['stat']['exists']:
                 logger.info("Collecting running golden config before test on {}".format(duthost.hostname))
                 duthost.shell("sonic-cfggen -d --print-data > /etc/sonic/running_golden_config.json")
-            duts_data[duthost.hostname]["pre_running_config"] = \
+            duts_data[duthost.hostname]["pre_running_config"]["host"] = \
                 json.loads(duthost.shell("cat /etc/sonic/running_golden_config.json", verbose=False)['stdout'])
 
+            if duthost.is_multi_asic:
+                for asic_index in range(0, duthost.facts.get('num_asic')):
+                    asic_ns = 'asic{}'.format(asic_index)
```

Comment:
Could use duthost.get_namespace_from_asic_id(asic_index) to avoid using magic string asic.

Reply:
We can't use get_namespace_from_asic_id, as this is relying on asics that are present in the card. For a chassis with not all fabric cards, the supervisor card will only have asics for the fabric cards that are present in the 'asics' variable. For asic_index of asics that are not present, we get an error